### ISSUE TITLE
Get Chromium unit_tests working

### ISSUE THREAD

**Issue by jonathanKingston** [brave/brave-browser#8376]
When running `npm run test -- unit_tests` I get the following:
```

> brave@1.6.34 test /home/jonathan/projects/brave-browser
> node ./scripts/commands.js test "unit_tests"

Recursing through GRD to find GRDP files...
Done recursing through GRD to find GRDP files.
/home/jonathan/projects/brave-browser/src: ninja -C /home/jonathan/projects/brave-browser/src/out/Component unit_tests
ninja: Entering directory `/home/jonathan/projects/brave-browser/src/out/Component'
[1/928] CXX obj/chrome/test/unit_tests/browsing_data_helper_unittest.o
[2/928] CXX obj/chrome/test/unit_tests/chrome_browser_main_extra_parts_metrics_unittest.o
[3/928] CXX obj/chrome/test/unit_tests/chrome_metrics_services_manager_client_unittest.o
[4/928] CXX obj/chrome/test/unit_tests/installable_task_queue_unittest.o
[5/928] CXX obj/chrome/test/unit_tests/subprocess_metrics_provider_unittest.o
[6/928] CXX obj/chrome/test/unit_tests/process_metrics_decorator_unittest.o
[7/928] CXX obj/chrome/test/unit_tests/browsing_data_local_storage_helper_unittest.o
[8/928] CXX obj/chrome/test/unit_tests/installable_manager_unittest.o
[9/928] CXX obj/chrome/test/unit_tests/non_recording_site_data_cache_unittest.o
[10/928] CXX obj/chrome/test/unit_tests/dns_probe_service_factory_unittest.o
[11/928] CXX obj/chrome/test/unit_tests/secure_dns_policy_handler_unittest.o
[12/928] CXX obj/chrome/test/unit_tests/browsing_data_media_license_helper_unittest.o
[13/928] CXX obj/chrome/test/unit_tests/browsing_data_quota_helper_unittest.o
[14/928] CXX obj/chrome/test/unit_tests/payment_handler_permission_context_unittest.o
[15/928] CXX obj/chrome/test/unit_tests/site_data_cache_factory_unittest.o
[16/928] CXX obj/chrome/test/unit_tests/managed_bookmarks_policy_handler_unittest.o
[17/928] CXX obj/chrome/test/unit_tests/media_stream_device_permission_context_unittest.o
[18/928] CXX obj/chrome/test/unit_tests/security_state_page_load_metrics_observer_unittest.o
[19/928] CXX obj/chrome/test/unit_tests/permission_context_base_feature_policy_unittest.o
[20/928] CXX obj/chrome/test/unit_tests/optimization_guide_permissions_util_unittest.o
[21/928] CXX obj/chrome/test/unit_tests/file_downloader_unittest.o
[22/928] CXX obj/chrome/test/unit_tests/navigation_predictor_unittest.o
[23/928] CXX obj/chrome/test/unit_tests/download_offline_content_provider_unittest.o
[24/928] CXX obj/chrome/test/unit_tests/push_messaging_app_identifier_unittest.o
[25/928] CXX obj/chrome/test/unit_tests/autocomplete_action_predictor_table_unittest.o
[26/928] CXX obj/chrome/test/unit_tests/persistent_notification_handler_unittest.o
[27/928] CXX obj/chrome/test/unit_tests/loading_predictor_config_unittest.o
[28/928] CXX obj/chrome/test/unit_tests/loading_data_collector_unittest.o
[29/928] CXX obj/chrome/test/unit_tests/push_messaging_service_unittest.o
[30/928] CXX obj/chrome/test/unit_tests/previews_page_load_metrics_observer_unittest.o
[31/928] CXX obj/chrome/test/unit_tests/services_unittest.o
[32/928] CXX obj/chrome/test/unit_tests/platform_notification_service_unittest.o
[33/928] CXX obj/chrome/test/unit_tests/tab_footprint_aggregator_unittest.o
[34/928] CXX obj/chrome/test/unit_tests/tab_specific_content_settings_unittest.o
FAILED: obj/chrome/test/unit_tests/tab_specific_content_settings_unittest.o 
/home/jonathan/projects/brave-browser/src/brave/script/redirect-cc.py ../../third_party/llvm-build/Release+Asserts/bin/clang++ -MMD -MF obj/chrome/test/unit_tests/tab_specific_content_settings_unittest.o.d -DBRAVE_CHROMIUM_BUILD -DDCHECK_ALWAYS_ON=1 -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_X11=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -DCR_CLANG_REVISION=\"n332890-c2443155-1\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DCOMPONENT_BUILD -D_LIBCPP_ABI_UNSTABLE -D_LIBCPP_ABI_VERSION=Cr -D_LIBCPP_ENABLE_NODISCARD -D_LIBCPP_DEBUG=0 -DCR_LIBCXX_REVISION=375504 -DCR_SYSROOT_HASH=bcc994cc6e5d4d6f0eec8b44e7f0a65f5a1a7b90 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DUSE_GIO -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DGTEST_API_= -DGTEST_HAS_POSIX_RE=0 -DGTEST_LANG_CXX11=1 -DGTEST_HAS_TR1_TUPLE=0 -DWEBP_EXTERN=extern -DVK_NO_PROTOTYPES -DGL_GLEXT_PROTOTYPES -DUSE_GLX -DUSE_EGL -DTOOLKIT_VIEWS=1 -DSYNC_PASSWORD_REUSE_DETECTION_ENABLED -DSYNC_PASSWORD_REUSE_WARNING_ENABLED -DON_FOCUS_PING_ENABLED -DEXPAT_RELATIVE_PATH -DBORINGSSL_SHARED_LIBRARY -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -DSK_HAS_PNG_LIBRARY -DSK_HAS_WEBP_LIBRARY -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\" -DSK_GL -DSK_HAS_JPEG_LIBRARY -DSK_USE_LIBGIFCODEC -DSK_VULKAN_HEADER=\"../../skia/config/SkVulkanConfig.h\" -DSK_VULKAN=1 -DSKIA_DLL -DSKCMS_API=__attribute__\(\(visibility\(\"default\"\)\)\) -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DVK_NO_PROTOTYPES -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -DPROTOBUF_USE_DLLS -DWEBRTC_ENABLE_SYMBOL_EXPORT -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0 -DWEBRTC_CHROMIUM_BUILD -DWEBRTC_POSIX -DWEBRTC_LINUX -DABSL_ALLOCATOR_NOTHROW=1 -DWEBRTC_USE_BUILTIN_ISAC_FIX=0 -DWEBRTC_USE_BUILTIN_ISAC_FLOAT=1 -DHAVE_SCTP -DNO_MAIN_THREAD_WRAPPING -DV8_USE_EXTERNAL_STARTUP_DATA -DLEVELDB_PLATFORM_CHROMIUM=1 -DLEVELDB_PLATFORM_CHROMIUM=1 -DLEVELDB_SHARED_LIBRARY -DUSING_V8_SHARED -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DI18N_ADDRESS_VALIDATION_DATA_URL=\"https://chromium-i18n.appspot.com/ssl-aggregate-address/\" -DUNIT_TEST -DSUPPORT_WEBGL2_COMPUTE_CONTEXT=1 -DWTF_USE_WEBAUDIO_PFFFT=1 -DUSE_LIBJPEG_TURBO=1 -DUSING_V8_SHARED -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DI18N_PHONENUMBERS_USE_ICU_REGEXP=1 -DI18N_PHONENUMBERS_USE_ALTERNATE_FORMATS=1 -DI18N_PHONENUMBERS_NO_THREAD_SAFETY=1 -DHUNSPELL_STATIC -DHUNSPELL_CHROME_CLIENT -DUSE_HUNSPELL -I../../brave/chromium_src -I../.. -Igen -I../../third_party/googletest/custom -I../../third_party/googletest/src/googletest/include -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -I../../third_party/libwebp/src -I../../third_party/khronos -I../../gpu -I../../third_party/vulkan/include -Igen/third_party/dawn/src/include -I../../third_party/dawn/src/include -I../../third_party/libyuv/include -I../../third_party/jsoncpp/source/include -I../../third_party/jsoncpp/generated -I../../third_party/boringssl/src/include -I../../third_party/ced/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -Igen -Igen -Igen -I../../third_party/skia -I../../third_party/libgifcodec -I../../third_party/vulkan/include -I../../third_party/skia/third_party/vulkanmemoryallocator -I../../third_party/vulkan/include -I../../third_party/protobuf/src -Igen/protoc_out -I../../third_party/protobuf/src -I../../third_party/webrtc_overrides -I../../third_party/webrtc -Igen/third_party/webrtc -I../../third_party/abseil-cpp -I../../third_party/crashpad/crashpad -I../../third_party/crashpad/crashpad/compat/non_mac -I../../third_party/crashpad/crashpad/compat/linux -I../../third_party/crashpad/crashpad/compat/non_win -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -Igen/third_party/perfetto -I../../third_party/libwebm/source -I../../third_party/leveldatabase -I../../third_party/leveldatabase/src -I../../third_party/leveldatabase/src/include -I../../v8/include -Igen/v8/include -Igen/third_party/metrics_proto -Igen/components/policy/proto -I../../third_party/re2/src -I../../third_party/mesa_headers -Igen -Igen -Igen -Igen -I../../third_party/libaddressinput/src/cpp/include -I../../third_party/cacheinvalidation/overrides -I../../third_party/cacheinvalidation/src -Igen -Igen -Igen -Igen -I../../third_party/googletest/custom -I../../third_party/googletest/src/googlemock/include -I../../third_party/flatbuffers/src/include -I../../third_party/libjpeg_turbo -I../../third_party/iccjpeg -I../../third_party/libpng -I../../third_party/zlib -I../../third_party/ots/include -I../../v8/include -Igen/v8/include -Igen -I../../third_party/libphonenumber/dist/cpp/src -Igen/third_party/libphonenumber -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -B../../third_party/binutils/Linux_x64/Release/bin -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -Xclang -mllvm -Xclang -instcombine-lower-dbg-declare=0 -fcomplete-member-pointers -m64 -march=x86-64 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -no-canonical-prefixes -Wall -Werror -Wextra -Wimplicit-fallthrough -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-bitwise-conditional-parentheses -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -O2 -fno-ident -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -ftrivial-auto-var-init=pattern -fvisibility=hidden -Xclang -add-plugin -Xclang find-bad-constructs -Xclang -plugin-arg-find-bad-constructs -Xclang check-ipc -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/libmount -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/blkid -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/glib-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/glib-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/lib/x86_64-linux-gnu/glib-2.0/include -Wno-shorten-64-to-32 -Wno-header-guard -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/nss -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/nspr -Wno-inconsistent-missing-override -Xclang -add-plugin -Xclang blink-gc-plugin -Xclang -plugin-arg-blink-gc-plugin -Xclang no-gc-finalized -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/gtk-3.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/at-spi2-atk/2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/at-spi-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/dbus-1.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/lib/x86_64-linux-gnu/dbus-1.0/include -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/gtk-3.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/gio-unix-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/cairo -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/libdrm -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/pango-1.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/harfbuzz -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/pango-1.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/fribidi -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/atk-1.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/cairo -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/pixman-1 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/uuid -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/freetype2 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/libpng16 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/gdk-pixbuf-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/libmount -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/blkid -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/glib-2.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem../../build/linux/debian_sid_amd64-sysroot/usr/include/dbus-1.0 -isystem../../build/linux/debian_sid_amd64-sysroot/usr/lib/x86_64-linux-gnu/dbus-1.0/include -std=c++14 -fno-exceptions -fno-rtti -nostdinc++ -isystem../../buildtools/third_party/libc++/trunk/include -isystem../../buildtools/third_party/libc++abi/trunk/include --sysroot=../../build/linux/debian_sid_amd64-sysroot -fvisibility-inlines-hidden -Wno-deprecated-declarations -c ../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc -o obj/chrome/test/unit_tests/tab_specific_content_settings_unittest.o
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:75:72: error: too few arguments to function call, expected at least 5, have 4
      origin, "A=B", base::Time::Now(), base::nullopt /* server_time */));
                                                                       ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:115:72: error: too few arguments to function call, expected at least 5, have 4
      origin, "C=D", base::Time::Now(), base::nullopt /* server_time */));
                                                                       ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:192:72: error: too few arguments to function call, expected at least 5, have 4
      origin, "A=B", base::Time::Now(), base::nullopt /* server_time */));
                                                                       ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:201:72: error: too few arguments to function call, expected at least 5, have 4
      origin, "C=D", base::Time::Now(), base::nullopt /* server_time */));
                                                                       ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:234:72: error: too few arguments to function call, expected at least 5, have 4
      origin, "A=B", base::Time::Now(), base::nullopt /* server_time */));
                                                                       ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:242:67: error: too few arguments to function call, expected at least 5, have 4
                                   base::nullopt /* server_time */));
                                                                  ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:265:77: error: too few arguments to function call, expected at least 5, have 4
                                             base::nullopt /* server_time */);
                                                                            ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:298:78: error: too few arguments to function call, expected at least 5, have 4
                                              base::nullopt /* server_time */);
                                                                             ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:301:57: error: too few arguments to function call, expected at least 5, have 4
      base::Time::Now(), base::nullopt /* server_time */);
                                                        ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:304:57: error: too few arguments to function call, expected at least 5, have 4
      base::Time::Now(), base::nullopt /* server_time */);
                                                        ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:307:57: error: too few arguments to function call, expected at least 5, have 4
      base::Time::Now(), base::nullopt /* server_time */);
                                                        ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
../../chrome/browser/content_settings/tab_specific_content_settings_unittest.cc:314:78: error: too few arguments to function call, expected at least 5, have 4
                                              base::nullopt /* server_time */);
                                                                             ^
../../brave/chromium_src/net/cookies/../../../../net/cookies/canonical_cookie.h:82:3: note: 'Create' declared here
  static std::unique_ptr<CanonicalCookie> Create(
  ^
12 errors generated.
[35/928] CXX obj/chrome/test/unit_tests/chrome_command_line_pref_store_unittest.o
[36/928] CXX obj/chrome/test/unit_tests/prerender_history_unittest.o
[37/928] CXX obj/chrome/test/unit_tests/permission_decision_auto_blocker_unittest.o
[38/928] CXX obj/chrome/test/unit_tests/chrome_pref_service_unittest.o
[39/928] CXX obj/chrome/test/unit_tests/browsing_data_shared_worker_helper_unittest.o
[40/928] CXX obj/chrome/test/unit_tests/proxy_policy_unittest.o
[41/928] CXX obj/chrome/test/unit_tests/content_settings_default_provider_unittest.o
[42/928] CXX obj/chrome/test/unit_tests/content_settings_pref_provider_unittest.o
[43/928] CXX obj/chrome/test/unit_tests/autocomplete_action_predictor_unittest.o
[44/928] CXX obj/chrome/test/unit_tests/profile_pref_store_manager_unittest.o
[45/928] CXX obj/chrome/test/unit_tests/chrome_client_connection_manager_unittest.o
[46/928] CXX obj/chrome/test/unit_tests/sound_content_setting_observer_unittest.o
[47/928] CXX obj/chrome/test/unit_tests/optimization_guide_hints_manager_unittest.o
[48/928] CXX obj/chrome/test/unit_tests/loading_predictor_unittest.o
[49/928] CXX obj/chrome/test/unit_tests/tab_load_tracker_unittest.o
[50/928] CXX obj/chrome/test/unit_tests/content_index_provider_unittest.o
[51/928] CXX obj/chrome/test/unit_tests/prerender_unittest.o
ninja: build stopped: subcommand failed.
null
```

See also: #8297

**Comment by jonathanKingston** [comment]
@bsclifton this is pretty much the same as the other, once compiling it still fails a lot. I haven't dug into if it is legitimate crashes or something that needs linking.

I see 171 test failures most of which are like this:
```
[ RUN      ] ExtensionServiceTest.ReloadAndReEnableExtension
Received signal 11 SEGV_MAPERR 000000000ad8
#0 0x7f7bc9d50869 base::debug::CollectStackTrace()
#1 0x7f7bc9c2b173 base::debug::StackTrace::StackTrace()
#2 0x7f7bc9d50321 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f7bbe6ed540 <unknown>
#4 0x560a030399d6 BraveBrowserProcessImpl::extension_whitelist_service()
#5 0x560a02f9ce9e extensions::BraveExtensionProvider::UserMayLoad()
#6 0x560a02917d02 extensions::ManagementPolicy::ApplyToProviderList()
#7 0x560a02917f98 extensions::ManagementPolicy::UserMayInstall()
#8 0x560a02919a32 extensions::PolicyCheck::Start()
#9 0x560a0291a408 extensions::PreloadCheckGroup::Start()
#10 0x560a04479695 extensions::UnpackedInstaller::StartInstallChecks()
#11 0x560a010c0e4b base::internal::Invoker<>::Run()
#12 0x7f7bc9cbc342 base::TaskAnnotator::RunTask()
#13 0x7f7bc9cda5f4 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#14 0x7f7bc9cda2ce base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoSomeWork()
#15 0x7f7bc9d8b904 base::MessagePumpLibevent::Run()
#16 0x7f7bc9cdb007 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#17 0x7f7bc9c95a57 base::RunLoop::Run()
#18 0x560a0517aa5f extensions::TestExtensionRegistryObserver::Wait()
#19 0x560a035e81b6 extensions::ChromeTestExtensionLoader::LoadUnpacked()
#20 0x560a035e6e74 extensions::ChromeTestExtensionLoader::LoadExtension()
#21 0x560a01a47f48 extensions::ExtensionServiceTest_ReloadAndReEnableExtension_Test::TestBody()
#22 0x560a026e0eb1 testing::Test::Run()
#23 0x560a026e1eef testing::TestInfo::Run()
#24 0x560a026e2697 testing::TestSuite::Run()
#25 0x560a026f2967 testing::internal::UnitTestImpl::RunAllTests()
#26 0x560a026f2422 testing::UnitTest::Run()
#27 0x560a0334a9d2 base::TestSuite::Run()
#28 0x560a033c53a8 content::UnitTestTestSuite::Run()
#29 0x560a03353e50 base::(anonymous namespace)::LaunchUnitTestsInternal()
#30 0x560a03353c99 base::LaunchUnitTests()
#31 0x560a0333a762 main
#32 0x7f7bbd7cf1e3 __libc_start_main
#33 0x560a003e004a _start
  r8: 0000000000000000  r9: 0000291a22f1d8c0 r10: 00007f7bba9f7860 r11: 0000000000000246
 r12: 0000000000000020 r13: 0000291a22ff6a50 r14: 0000291a23283a78 r15: 0000291a22f1d8c0
  di: 0000000000000000  si: 0000291a22ff6a50  bp: 00007ffc73934ae0  bx: 0000291a22fc2ac8
  dx: 000000000000006c  ax: 0000560a05b37410  cx: 0000000000000000  sp: 00007ffc73934ad0
  ip: 0000560a030399d6 efl: 0000000000010246 cgf: 002b000000000033 erf: 0000000000000004
 trp: 000000000000000e msk: 0000000000000000 cr2: 0000000000000ad8
[end of stack trace]
Calling _exit(1). Core file will not be generated.
[11890/12279] ExtensionServiceTest.ReloadAndReEnableExtension (CRASHED)
```

I'll upload the patch now again.

**Comment by jonathanKingston** [comment]
Initial findings:

- It seems in many of these tests that `g_browser_process` isn't initialised in the right order that Brave code expects it / not at all.
  - I was able to fix some of these just by checking if the process existed; however there might be more fundamental lifecycle issues here.
- I'm seeing a lot of the same error reported in: #2125 it looks to me like the ads service isn't initialised in the right order always too.

**Comment by jonathanKingston** [comment]
It seems that many things are missing from the compiled sources of unit_tests when tabs are created, it seems like there aren't brave unit tests for this but it should be spinning up the ads service etc like some of the tests do for brave tests.

I'm working on adding a similar set of dependencies to the chromium test `BUILD.gn` files and the mock service stuff in `brave/components/brave_ads/browser/test_util.cc` to `src/chrome/test/base/testing_profile.cc`

**Comment by jonathanKingston** [comment]
> It seems in many of these tests that g_browser_process isn't initialised in the right order that Brave code expects it / not at all. 

It's actually not this, `g_brave_browser_process` isn't initialised due to the tests using `TestingBrowserProcess` which doesn't extend from `BraveBrowserImpl` and instead `BrowserProcess` which means it can't simply be swapped out.

I tried extending `BraveBrowserImpl` instead and initialising various parts, but it's not a simple transposition and I'm starting to think moving the brave code into an extension of `BrowserProcess` rather than `BrowserProcessImpl` would be simpler than fixing `TestingBrowserProcess`.

**Comment by mariospr** [comment]
I'm starting to look into this in between rebases work as per a conversation with @bbondy on Wednesday, and just wanted to dump here the current situation (as of Brave 1.22.6, based on Chromium 89.0.4389.40), as I understand it:
  * Currently `master` is failing to build `unit_tests` due to just 2 errors, being addressed in https://github.com/brave/brave-core/pull/7850.
  * Doing a first run of `npm run test -- unit_tests Debug --test-launcher-timeout=100000` without touching anything else yields a **88.16% passing rate**, with the following results ([more detail in this gist](https://gist.github.com/mariospr/5e2aa79620c0d0d46adb921229b82686)):
    * **Total tests run**: 15715
      * **Passed**: 13854
      * **Failures**: 1861 (168 failed, 1693 crashed)

Talking to @bridiver, he made a point (which I agree with) on that we shouldn't aim for having all Chromium unit tests working, since Brave intentionally changes Chromium behaviour in ways that will obviously break those tests... and so we shouldn't try to fix them, but rely on Brave's own unit tests to check that particular "conflicting" functionality.

However, there's still value on making sure that the tests from Chromium that should be passing (which, from the passing rate above, is not a small number!), or at least a subset of them, keep passing fine regardless of changes happening in the brave-core repository, in order to improve test coverage and catch unintentionally errors more easily and faster.

With this in mind, it seems to me that the way forward here could be summarized in 3 steps:
  1. Get `unit_tests` building fine (re: https://github.com/brave/brave-core/pull/7850)
  2. Find a way to filter out the unit_tests from Chromium that we're currently not interested in running, and remove them (e.g. via `sources -= <test file>` rules) from building, so that we don't have to patch them at all.
  3. Once we get to a point where we have a subset of `unit_tests` where all are passing for Brave build, we could try to get a deeper look into those that we skipped in step (2) and see if there's something test that we should do an extra effort to keep running them, ideally with minimal patching or not patching at all.

I believe that for now we could focus on steps (1) and (2) and, even if we just did that and ended up with a a lot of tests disabled because they belong to files we don't want to patch, I think that could already be a net positive for Brave. That is, even if we only manage to enable, say, about half of the 15715 tests, that would still mean increasing test coverage in ~7500 tests, which I think it's worth exploring. Later on, and depending on other variables and constraints, we could consider (3) as well, in order to extend that test coverage even further.

And if this approach seems sensible and ends up working out not too badly, I suppose we could expand it to https://github.com/brave/brave-browser/issues/8297, and do the same thing for Chromium's `browser_tests`

@bbondy @bridiver (+ others) WDYT?
